src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/ImmutableDispatchRuntime.cs

```diff
@@ -550,6 +550,7 @@ internal async Task<MessageRpc> ProcessMessageAsync(MessageRpc rpc)
                 rpc = await _authenticationBehavior.AuthenticateAsync(rpc);
             }
 
+            rpc.Operation.SetClaimsPrincipalToOperationContext(rpc);
             if (_authorizationBehavior != null)
```

Comment:
This should be wrapped in `if (Parent.RequireClaimsPrincipalOnOperationContext)` otherwise the call might throw. This means you need to make sure that whatever mechanism sets `RequireClaimsPrincipalOnOperationContext` to true is triggered in this scenario.

Reply:
done here https://github.com/CoreWCF/CoreWCF/pull/850/commits/bc84c9202bbc8cb1df0ca6e430366e599afa4e35